tutorials/run-spring-petclinic-on-app-engine-flex-cloudsql.md

```diff
@@ -0,0 +1,131 @@
+
+### Prerequisites
+1. Create a project in the [Google Cloud Platform Console](https://console.cloud.google.com/).
+1. Enable billing for your project.
+1. Install the [Google Cloud SDK](https://cloud.google.com/sdk/).
+
+### Prepare
+1. Initialize, create an app engine application and authorize `gcloud` for using GCP APIs in your local environment :
+```
+   gcloud init
+   gcloud app create
+   gcloud auth application-default login
+```
+1. Clone and test the Pet Clinic application locally :
+```
+    git clone https://github.com/spring-projects/spring-petclinic.git
+    cd spring-petclinic
+    ./mvnw spring-boot:run
+```
+
+  Access http://localhost:8080 via your web browser to see the application homepage.
+
+  By default, the application uses an [hsqldb](http://hsqldb.org/), an in-memory database.
+  We will now switch from `hsqldb` to using CloudSQL as your database.
+
+### Using CloudSQL as your database
+
+1. Enable the [Cloud SQL API](https://console.cloud.google.com/flows/enableapi?apiid=sqladmin).
+
+1. Create a 2nd Generation CloudSQL (MySQL) instance and set the root user password following the instructions [here](https://cloud.google.com/sql/docs/mysql/create-instance#create-2nd-gen).
+
+1. Create the `petclinic` database.
+```
+    gcloud beta sql databases create petclinic --instance=INSTANCE_NAME
+```
+1. Note the `connectionName` of the instance in the format `project-id:zone-id:instance-id`  using
+```
+   gcloud beta sql instances describe INSTANCE_NAME
+```
+1. Update `src/main/resources/application-mysql.properties`, replacing INSTANCE_CONNECTION_NAME with the `connectionName` from the previous step.
+```
+    database=mysql
+    spring.datasource.driverClassName=com.mysql.jdbc.Driver
+    spring.datasource.url=jdbc:mysql://google/petclinic?cloudSqlInstance=INSTANCE_CONNECTION_NAME&socketFactory=com.google.cloud.sql.mysql.SocketFactory
+    spring.datasource.username=root
+    spring.datasource.password=my-smart-password
+```
+1. Update `pom.xml` to include the [Cloud SQL MySQL Socket Factory socket library](https://github.com/GoogleCloudPlatform/cloud-sql-mysql-socket-factory).
+```
+    <dependency>
+        <groupId>com.google.cloud.sql</groupId>
+        <artifactId>mysql-socket-factory</artifactId>
+        <version>1.0.2</version>
+    </dependency>
+```
+   The socket library allows to connect to your Cloud SQL instance for local testing and in deployment.
+1. Restart the SpringBoot application using the sql [profile](http://docs.spring.io/spring-boot/docs/current/maven-plugin/examples/run-profiles.html)
+```
+   ./mvnw -Drun.profiles=mysql spring-boot:run
+```
+  Access the application homepage http://localhost:8080 via your web browser and add some data.
+1. You can verify the data exists in CloudSQL running queries agains the `petclinic` database using the [Cloud shell](https://cloud.google.com/sql/docs/mysql/quickstart#connect_to_your_instance_using_the_db_client_client_in_the_cloud_shell)
```

Comment:
s/CloudSQL running/Cloud SQL by running/

also, s/[Cloud shell]/[Cloud Shell]/

also, add a period to the end of the sentence